src/OrchardCore.Modules/OrchardCore.Lists/OrchardCore.Lists.csproj

```diff
@@ -25,6 +25,7 @@
     <ProjectReference Include="..\..\OrchardCore\OrchardCore.Contents.Core\OrchardCore.Contents.Core.csproj" />
     <ProjectReference Include="..\..\OrchardCore\OrchardCore.Contents.TagHelpers\OrchardCore.Contents.TagHelpers.csproj" />
     <ProjectReference Include="..\..\OrchardCore\OrchardCore.ContentTypes.Abstractions\OrchardCore.ContentTypes.Abstractions.csproj" />
+    <ProjectReference Include="..\..\OrchardCore\OrchardCore.Data.YesSql\OrchardCore.Data.YesSql.csproj" />
```

Comment:
> I think OrchardCore.Data.Abstractions, as an abstraction layer, should not rely on a specific implementation.

Also OrchardCore modules should not rely on a specific implementation

Reply:
They should not, but many of them do.

The PR removes the dependency of `OrchardCore.Data.Abstractions` on `YesSql.Core`.

Because of that, actually many modules in OC has an indirect dependency on `YesSql`, even if not needed.

DataMigration which is a core feature directly relies on YesSql because of `ISchemaBuilder` property which is a `YesSql` interface.

This PR does not fix all dependency problems over `YesSql`, it's just a first step to isolate and identify which module relies on `YesSql`. There is much more work to create all necessary abstraction layers to properly decouple modules from infrastructure.
But I think it's acceptable to have `YesSql` dependency for everything that is related to content management, but not for modules that could be used in a different context.

Reply:
IMHO some of the YesSQL primitive was to be moved into `OrchardCore.Data.Abstractions`, so almost the modules will rely on it not in YesSQL

> The PR removes the dependency of OrchardCore.Data.Abstractions on YesSql.Core

This is a very good step